pkg/controllers/dashboard/plugin/controller.go

```diff
@@ -84,6 +109,8 @@ func (h *handler) OnPluginChange(key string, plugin *v1.UIPlugin) (*v1.UIPlugin,
 			_, err2 := h.plugin.Update(p)
 			if err2 != nil {
 				logrus.Errorf("failed to update plugin [%s] noCache flag: %s", p.Spec.Plugin.Name, err2.Error())
+				p.Status.Ready = false
```

Comment:
From what I could gather, it does it so that if a plugin failed to cache before, it would try again even if the next trigger is not for it. If all plugins but the one we receive have already been cached, it will skip as seen [here](https://github.com/rancher/rancher/blob/00b5fa48960f886c84a93a5063431263716a9752/pkg/controllers/dashboard/plugin/fscache.go#L58). 

Now, that being said, I'm not entirely sure if this is useful since the controller already retries by itself.